dut/rocket-chip-dut.wake

```diff
@@ -72,20 +76,31 @@ global def rocketChipDUTMaker dutPlan userArgs =
   def allOutputs = rocketOutputs.getRocketChipGeneratorOutputsAllOutputs
   def topModuleOpt = dutPlan.getRocketChipDUTPlanTestharness.getTestharnessTopModule
   def dts = rocketOutputs.getRocketChipGeneratorOutputsDTS
+
+  # TODO: This should be something that is captured by
+  # RocketChipGeneratorOutputs in rocket-chip.git, since only the Wake rules in
+  # that repo can know what the final output filenames look like.
+  # This can be replaced once we actually add that in rocket-chip.git.
   def dtsJSON =
-    def fileName =
+    def fileNameOpt =
       rocketOutputs
       | getRocketChipGeneratorOutputsInputOptions
-      | getRocketChipGeneratorOptionsConfigNames
-      | catWith "_"
-      | ("_root_.{_}.json")
-    def fileRegex = regExpCat (`([^/]*/)*`, fileName.quote, Nil)
-
-    rocketOutputs
-    | getRocketChipGeneratorOutputsAllOutputs
-    | find (matches fileRegex _.getPathName)
-    | omap getPairFirst
-    | getOrElse "{fileName} file not found".makeError.makeBadPath
+      | getRocketChipGeneratorOptionsBaseFileName
+      | omap ("{_}.json")
+
+    def getDTSJSONForFileName fileName =
+      def fileRegex = regExpCat (`([^/]*/)*`, fileName.quote, Nil)
+
+      rocketOutputs
+      | getRocketChipGeneratorOutputsAllOutputs
+      | find (matches fileRegex _.getPathName)
+      | omap getPairFirst
+      | getOrFail "{fileName} file not found".makeError
```

Comment:
I think this is a better pattern. this way you propagate the original error message if the job failed. and also prevent matching on the path name `BAD_PATH`
```
outputs
| findFailFn getPathResult
| rmapPass (
  _
  | find (matches fileRegex _.getPathName)
  | omap getPairFirst
  | getOrFail "{fileName} file not found".makeError
)

Reply:
Let me double check that I understand the difference between what I had and what you wrote. In your approach, you first filter the job for all and any failed paths prior to trying to extract the desired path out of the outputs list, which means that we can error out earlier if the whole thing failed. In my approach, the `find` expression would merely fail to match anything because the BadPaths would not have real PathNames, and it would fall through to the "file not found" error. Is this correct?

cc @terpstra as an example of where this pattern can lead to intuitive and awkward results.

I'll go with your approach, @albertchen-sifive.

Reply:
yes that's correct

Reply:
Might have spent a bit too long figuring out how to write this, but I think I have this factored in a way that I like and that does not rely on nested pipelines: https://github.com/sifive/api-generator-sifive/pull/49/commits/4bc83a79cfae82e77f2070e844755867d1fce4b0